SConstruct

```diff
@@ -179,18 +177,43 @@ for k in platform_opts.keys():
     for o in opt_list:
         opts.Add(o)
 
-for x in module_list:
-    module_enabled = True
-    tmppath = "./modules/" + x
-    sys.path.insert(0, tmppath)
+# Detect modules.
+opts.Add("custom_modules", "Directory path containing custom modules to build along with built-in modules.", "")
+
+modules_detected = {}
+module_search_paths = ["modules"]  # Built-in path.
+
+if ARGUMENTS.get("custom_modules"):
+    paths = ARGUMENTS.get("custom_modules").split(os.pathsep)
```

Comment:
```suggestion
    paths = ARGUMENTS.get("custom_modules").split(":")
```
I wonder whether it makes sense to rely on platform-specific path separator at all, yet that's not too critical.

Reply:
If you want to specify an absolute path on Windows, you have to use `:` to specify the drive letter. This is why `;` should be used as a path separator there.

Reply:
Ah yes makes sense, it's just that I've seen Godot uses some double colon `::` in paths when running from command line, but this wouldn't be intuitive.